homeassistant/components/ozw/lock.py

```diff
@@ -37,3 +75,48 @@ async def async_lock(self, **kwargs):
     async def async_unlock(self, **kwargs):
         """Unlock the lock."""
         self.values.primary.send_value(False)
+
+    def set_usercode(self, code_slot, usercode):
+        """Set the usercode to index X on the lock."""
+        lock_node = self.values._node.values()
+
+        for value in lock_node:
+            if (
+                value.command_class == CommandClass.USER_CODE
+                and value.index == code_slot
+            ):
+                if len(str(usercode)) < 4:
+                    _LOGGER.error(
+                        "Invalid code provided: (%s) user code must be at least 4 digits",
+                        usercode,
+                    )
+                    break
+                value.send_value(usercode)
+                break
+
+    def get_usercode(self, code_slot):
```

Comment:
This seems like a weird service. Actually thinking now, why would any of these be services? This seems more like configuration and the user can change this with the OZW Admin panel?

Reply:
I believe they are services so that they can be run through automatons. IE: Enable Code "5312" every tuesday between 2pm-5pm then disable it.r

Reply:
Correct it's for automations and multiple lock manager packages that have been created.